Do not concat Bundle-Activator when already defined

chrisrueger opened the pull request:
Closes #6867

fixes a problem that `Verifier.verifyActivator()` throws an error "The Bundle-Activator header only supports a single type. The following types were found" when the bnd instructions already define a Bundle-Activator but a processed jar also contributed a Bundle-Activator (e.g. via `@Header(name = Constants.BUNDLE_ACTIVATOR, value = "${@class}")` annotation). 

see [org.apache.logging.log4j.util.Activator](https://github.com/apache/logging-log4j2/blob/2.x/log4j-api/src/main/java/org/apache/logging/log4j/util/Activator.java#L49)

<img width="1826" height="436" alt="image" src="https://github.com/user-attachments/assets/1278883c-39bb-4d0a-9e4a-6df249185e9f" />


Before this fix the Bundle-Activator header was concatenated to the existing `Bundle-Activator` header in the outer bnd of the bundle and thus resulted in multiple values (separated by comma). But for Bundle-Activator this does not make sense as it supports only a single value, as there can only be one single Bundle-Activator in the manifest - which is correctly complained about by `Verifier.verifyActivator()`.

This PR fixes the root cause where the multiple-values are created, so that `Verifier.verifyActivator()` does not complain anymore.

This for example happened with `-includeresource` of a Bundle (unrolling or lib:=true) or other ways where bnd scans classes of jar files.

I also identified some other Bundle headers in which should also only contain a single value and not be concatenated.

chrisrueger commented:
Note: A related issue which touches  a similar topic https://github.com/bndtools/bnd/issues/5363 (and non-merged PR https://github.com/bndtools/bnd/pull/5389 )

stbischof commented:
+1

juergen-albert commented:
The Mechanism should also make sure, that any bundle activator header is not just a singlton, but that the property in the bnd file wins, so I can overwrite whatever included classfiles may define here.

chrisrueger commented:
I renamed the private variable to `AnnotationHeaders.BUNDLE_HEADERS_ANALYZER_WINS` and added some explanation. I hope this makes the intention clearer. Since it's all private people can come up with better wording in the future.

stbischof commented:
Here is an example where i would use this 

https://github.com/eclipse-osgi-technology/command/blob/main/all/src/main/java/org/eclipse/osgi/technology/command/all/Activator.java#L26
